examples/carousel/js/carousel-tablist.js

```diff
@@ -0,0 +1,373 @@
+
+  var tabpanelNode = this.getTabpanelNode(tabNode);
+
+  if (tabpanelNode) {
+    tabpanelNode.classList.remove('active');
+  }
+}
+
+CarouselTablist.prototype.showTabpanel = function (tabNode, moveFocus) {
+  if (typeof moveFocus !== 'boolean') {
+    moveFocus = false;
+  }
+
+  tabNode.setAttribute('aria-selected', 'true');
+  tabNode.removeAttribute('tabindex');
+
+  var tabpanelNode = this.getTabpanelNode(tabNode);
+
+  if (tabpanelNode) {
+    tabpanelNode.classList.add('active');
+  }
+
+  if (moveFocus) {
+    tabNode.focus();
+  }
+}
+
+CarouselTablist.prototype.setSelectedTab = function (newTab, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  for (var i = 0; i < this.tabNodes.length; i++) {
+    this.hideTabpanel(this.tabNodes[i]);
+  }
+
+  this.currentTabNode = newTab;
+  this.showTabpanel(newTab, moveFocus);
+}
+
+CarouselTablist.prototype.setSelectedToPreviousTab = function (currentTabNode, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  var index;
+
+  if (typeof currentTabNode !== 'object') {
+    currentTabNode = this.currentTabNode;
+  }
+
+  if (currentTabNode === this.firstTabNode) {
+    this.setSelectedTab(this.lastTabNode, moveFocus);
+  }
+  else {
+    index = this.tabNodes.indexOf(currentTabNode);
+    this.setSelectedTab(this.tabNodes[index - 1], moveFocus);
+  }
+}
+
+CarouselTablist.prototype.setSelectedToNextTab = function (currentTabNode, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  var index;
+
+  if (typeof currentTabNode !== 'object') {
+    currentTabNode = this.currentTabNode;
+  }
+
+  if (currentTabNode === this.lastTabNode) {
+    this.setSelectedTab(this.firstTabNode, moveFocus);
+  }
+  else {
+    index = this.tabNodes.indexOf(currentTabNode);
+    this.setSelectedTab(this.tabNodes[index + 1], moveFocus);
+  }
+}
+
+CarouselTablist.prototype.setSelectedTofirstTabNode = function (moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+  this.setSelectedTab(this.firstTabNode, moveFocus);
+}
+
+CarouselTablist.prototype.setSelectedTolastTabNode = function (moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+  this.setSelectedTab(this.lastTabNode, moveFocus);
+}
+
+CarouselTablist.prototype.rotateSlides = function () {
+  if (this.rotate) {
+    this.setSelectedToNextTab();
+  }
+  setTimeout(this.rotateSlides.bind(this), this.timeInterval);
+}
+
+CarouselTablist.prototype.updateRotation = function () {
+
+  // Pause rotation when keyboard focus enters the carousel
+  if (this.hasFocus) {
+    this.isStopped = true;
+  }
+
+  if (!this.hasHover && !this.hasFocus && !this.isStopped) {
+    this.rotate = true;
+    this.liveRegionNode.setAttribute('aria-live', 'off');
+  }
+  else {
+    this.rotate = false;
+    this.liveRegionNode.setAttribute('aria-live', 'polite');
+  }
+
+  if (this.isStopped) {
```

Comment:
This logic still confuses me -- why is the play/pause button updated when rotation is stopped by focus, but allowed to show the opposite icon and label when the carousel is hovered?